src/TSTransformer/nodes/transformSourceFile.ts

```diff
@@ -205,13 +205,26 @@ export function transformSourceFile(state: TransformState, node: ts.SourceFile)
 		}
 	}
 
+	const headerStatements = luau.list.make<luau.Statement>();
+
+	// add build information to the tree
+	luau.list.push(headerStatements, luau.comment(` Compiled with roblox-ts v${COMPILER_VERSION}`));
+
 	// add the Runtime library to the tree if it is used
 	if (state.usesRuntimeLib) {
-		luau.list.unshift(statements, state.createRuntimeLibImport(node));
+		luau.list.push(headerStatements, state.createRuntimeLibImport(node));
 	}
 
-	// add build information to the tree
-	luau.list.unshift(statements, luau.comment(`Compiled with roblox-ts v${COMPILER_VERSION}`));
+	// extract Luau directive comments like --!strict so we can put them before headerStatements
+	const directiveComments = luau.list.make<luau.Statement>();
+	while (statements.head && luau.isComment(statements.head.value) && statements.head.value.text.startsWith("!")) {
+		const comment = luau.list.shift(statements);
+		assert(comment, "Expected comment statement!");
```

Comment:
This assert could just be a non-null assertion, since statements.head is part of the while condition. As with most type assertions I would recommend putting a comment alongside it to explain the safety, eg `// safety: statements.head is checked in while condition`

Reply:
> This assert could just be a non-null assertion
Do you mean `assert(comment !== undefined);`?

Reply:
No, I mean `comment!`